extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java

```diff
@@ -223,6 +223,15 @@ public class ArcConfig {
     @ConfigItem
     public ArcContextPropagationConfig contextPropagation;
 
+    /**
+     * If set to {@code true}, the container should try to optimize the contexts for some of the scopes.
```

Comment:
If we want to keep this documented and exposed, we should probably be more accurate in what it does and what scopes exactly are affected so that it's understandable for users.
But personally, I would rather choose an undocumented version that we eventually completely remove if we don't see any issues with this approach.

Reply:
Let's keep it hidden.